src/goto-instrument/contracts/contracts.cpp

```diff
@@ -682,6 +788,7 @@ void code_contractst::instrument_call_statement(
       assigns,
       instruction_it,
       pointer_object(instruction_it->call_arguments().front()));
+    add_pragma_disable_assigns_check(*instruction_it);
```

Comment:
Same as above.

Reply:
The idea is to not instrument the same instruction again. Do you think it's worth adding a comment on that?

Reply:
I moved this line to the `for` loop that iterated over all instructions, instead of repeating it in all `instrument_*` functions, and added a clarifying comment.